src/bootstrap/format.rs

```diff
@@ -191,10 +169,21 @@ pub fn format(build: &Builder<'_>, check: bool, paths: &[PathBuf]) {
                 ignore_fmt.add(&format!("!/{}", untracked_path)).expect(&untracked_path);
             }
             if !check && paths.is_empty() {
-                if let Some(files) = get_modified_rs_files(build) {
-                    for file in files {
-                        println!("formatting modified file {file}");
-                        ignore_fmt.add(&format!("/{file}")).expect(&file);
+                match get_modified_rs_files(build) {
+                    Ok(Some(files)) => {
+                        for file in files {
+                            println!("formatting modified file {file}");
+                            ignore_fmt.add(&format!("/{file}")).expect(&file);
+                        }
+                    }
+                    Ok(None) => {}
+                    Err(err) => {
+                        println!(
+                            "WARN: Something went wrong when running git commands:\n{err}\n\
+                            Falling back to formatting all files."
+                        );
+                        // Something went wrong when getting the version. Just format all the files.
+                        paths.push(".".into());
```

Comment:
I've started at this for quite a while now, but I don't think I get it... why does this push `.` to the path list? We have other error cases below ("could not find usable git"), and they don't do that. Why is this different? I think for `./x.py fmt` it is unnecessary since we're anyway searching everything, and for `./x.py fmt compiler/` this s actively buggy since it formats not just the files in the compiler dir. Am I missing something?

Reply:
you're probably right. i don't remember the details of this change though, but this does look sus